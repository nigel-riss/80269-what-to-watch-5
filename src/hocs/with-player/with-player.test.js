import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import withPlayer from './with-player.jsx';


const MockComponent = (props) => {
  const {
    children,
  } = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withPlayer(MockComponent);


it(`Should withPlayer render correctly`, () => {
  const tree = renderer
    .create((
      <MockComponentWrapped>
        <React.Fragment/>
      </MockComponentWrapped>
    ))
    .toJSON();

  expect(tree).toMatchSnapshot();
});
