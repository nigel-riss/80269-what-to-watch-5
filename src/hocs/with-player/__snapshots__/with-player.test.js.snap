// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should withPlayer render correctly 1`] = `<div />`;
